OpenSim/Simulation/Model/Model.cpp

```diff
@@ -974,6 +974,16 @@ void Model::extendConnectToModel(Model &model)
 
     // TODO: Get rid of the SimbodyEngine
     updSimbodyEngine().connectSimbodyEngineToModel(*this);
+
+    // Cache enabled controllers locally so that `computeControls` can
+    // iterate over the cache, rather than needing to call
+    // `getComponentList` (expensive) on each iteration.
+    this->_enabledControllers.clear();
+    for (const Controller& controller : getComponentList<Controller>()) {
+        if (controller.isEnabled()) {
```

Comment:
The problem is in `Controller::isEnabled()` which should not be referencing the `Model` to figure if it is enabled or not since it has its own`enabled` property and  should just return `get_enabled()` .